keys.go

```diff
@@ -217,11 +229,11 @@ func (c *Client) createKey(ctx context.Context, key Key) (*Key, error) {
 		Keys: []Key{key},
 	}
 
-	req, err := c.newRequest("POST", "keys", &keysRequest)
+	req, err := c.newRequest(http.MethodPost, path, &keysRequest)
 	if err != nil {
 		return nil, err
 	}
-
+	req.Header.Set("Prefer", preferHeaders[ReturnRepresentation])
```

Comment:
Initially the user always got minimal representation. The only way the user could get full representation was through a `GetKey` .The reason we might want to return full representation is that in minimal representation policy details is not returned in response. I believe, we definitely want full representation for `KeyWithPolicyOverrides` functions as it returns the policy details that the user needs to see. For Create Key flow also, I dont think this would create any issue to users as we are just giving back more information. Let me know what you think.

Reply:
So Ankit, what I have done for now is add full representation only for `CreateKeyWithPolicyOverrides` functionality

Reply:
I think if the user wants the perfer header to be minimalRepresentation then we should return the minimalRepresentation only instead of full representation.

Any suggestion here @dinesh-venkatraman-g